jmxtrans-core/src/main/java/com/googlecode/jmxtrans/guice/JmxTransModule.java

```diff
@@ -83,7 +87,7 @@ protected void configure() {
 				.toInstance(getObjectPool(new SocketFactory(), SocketFactory.class.getSimpleName()));
 		bind(new TypeLiteral<GenericKeyedObjectPool<SocketAddress, DatagramSocket>>(){})
 				.toInstance(getObjectPool(new DatagramSocketFactory(), DatagramSocketFactory.class.getSimpleName()));
-		bind(KeyedObjectPool.class).annotatedWith(Names.named("mbeanPool"))
+		bind(new TypeLiteral<KeyedObjectPool<JmxConnectionProvider, JMXConnection>>(){}).annotatedWith(Names.named("mbeanPool"))
```

Comment:
This change does not seem to be related to the title. Maybe split it in another PR?